src/App.js

```diff
@@ -63,36 +63,60 @@ export default function App() {
-          onCloseConfirmationModal={closeModalConfirmationHandler}
-          onShowConfirmationHandler={showModalConfirmationHandler}
-          onDeleteActivity={deleteActivity}
-          showModalConfirmation={showModalConfirmation}
+      <Header />
+      <Routes>
+        <Route
+          path="/"
+          element={
+            <ActivityPage
+              activities={filteredActivities}
+              onSetActivities={setActivities}
+              onToggleFavorites={toggleFavorites}
+              onCloseConfirmationModal={closeModalConfirmationHandler}
+              onShowConfirmationHandler={showModalConfirmationHandler}
+              onDeleteActivity={deleteActivity}
+              showModalConfirmation={showModalConfirmation}
+              onFilterTypeReset={filterTypeReset}
+              onFilterTypeValue={setTypeFiltervalue}
+              onFilterDurationReset={filterDurationReset}
+              onFilterDurationValue={setDurationFilterValue}
+            />
+          }
         />
-      )}
-      {currentPage === 'favorites' && (
-        <FavoritesPage
-          activities={activities}
-          onNavigate={page => setCurrentPage(page)}
-          onToggleFavorites={toggleFavorites}
-          onCloseConfirmationModal={closeModalConfirmationHandler}
-          onShowConfirmationHandler={showModalConfirmationHandler}
-          onDeleteActivity={deleteActivity}
-          showModalConfirmation={showModalConfirmation}
+        <Route
+          path="/activities"
+          element={
+            <ActivityPage
+              activities={filteredActivities}
+              onSetActivities={setActivities}
+              onToggleFavorites={toggleFavorites}
+              onCloseConfirmationModal={closeModalConfirmationHandler}
+              onShowConfirmationHandler={showModalConfirmationHandler}
+              onDeleteActivity={deleteActivity}
+              showModalConfirmation={showModalConfirmation}
+              onFilterTypeReset={filterTypeReset}
+              onFilterTypeValue={setTypeFiltervalue}
+              onFilterDurationReset={filterDurationReset}
+              onFilterDurationValue={setDurationFilterValue}
+            />
+          }
         />
-      )}
-      {currentPage === 'map' && <Map onSetIsHidden={() => setCurrentPage('activity')} />}
+        <Route
+          path="/favorites"
+          element={
+            <FavoritesPage
+              activities={activities}
+              onToggleFavorites={toggleFavorites}
+              onCloseConfirmationModal={closeModalConfirmationHandler}
+              onShowConfirmationHandler={showModalConfirmationHandler}
+              onDeleteActivity={deleteActivity}
+              showModalConfirmation={showModalConfirmation}
+            />
+          }
+        />
+        <Route path="/map" element={<Map />} />
+        <Route path="/addActivity" element={<AddActivityPage onSetActivities={setActivities} />} />
```

Comment:
Same here about the `setActivities` state!

Reply:
Changed this as well. named it setActivityHandler. I was just experimenting... It's working now but i don't really know if that's what you're expecting and trying to tell me - we'll see 😄